AK/Utf8View.h

```diff
@@ -218,6 +218,8 @@ class Utf8View {
 
     static constexpr bool is_valid_code_point(u32 code_point, size_t byte_length)
     {
+        if (byte_length == 3 && code_point >= 0xD800 && code_point <= 0xDFFF)
```

Comment:
Is this check okay? This function is pretty hot so maybe this branch is deserving of an [[unlikely]]. I'm not really sure though because I haven't tested the performance hit from this condition.

Reply:
This isn't right - there's nothing inherently wrong about UTF-8 encoded bytes decoding to U+D8000. That's entirely a UTF-16 restriction.

Reply:
[Rust validates UTF8 this way](https://doc.rust-lang.org/src/core/str/validations.rs.html#162) and the [Wasm spec](https://github.com/WebAssembly/testsuite/blob/6dfedc8b8423a91c1dc340d3af1a7f4fbf7868b4/utf8-custom-section-id.wast#L453) also does to. 

In the [spec](https://datatracker.ietf.org/doc/html/rfc3629#page-5), it says
> The definition of UTF-8 prohibits encoding character numbers between U+D800 and U+DFFF, which are reserved for use with the UTF-16 encoding form (as surrogate pairs) and do not directly represent characters

Reply:
Huh weird - TIL. It causes some test failures though so something in the system is relying on this.

Also while I'm here, a couple things about the patch:
* Such a change should come with a test in Tests/AK/TestUtf8.cpp
* These are really 2 separate changes and thus should be 2 separate commits

Reply:
Yeah I also learned it today, pretty strange. Should I make a separate PR for it? Or just another commit? I'll fix the test failures and investigate what in the system is reliant on the alternative behavior.

Reply:
Just a separate commit is fine :)

Reply:
Sounds good!